src/Test/L0/Worker/Release/FetchEngineL0.cs

```diff
@@ -0,0 +1,187 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.Services.Agent.Worker.Release;
+using Microsoft.VisualStudio.Services.Agent.Worker.Release.ContainerFetchEngine;
+using Moq;
+using Xunit;
+
+namespace Microsoft.VisualStudio.Services.Agent.Tests.Worker.Release
+{
+    public sealed class FetchEngineL0
+    {
+        private readonly IEnumerable<ContainerItem> mockContainerItems = new List<ContainerItem>
+        {
+            DummyConatinerItem1,
+            DummyConatinerItem2
+        };
+
+        private readonly byte[] mockItemContent = Encoding.UTF8.GetBytes("Item Content");
+
+        private readonly ContainerFetchEngineOptions containerFetchEngineTestOptions = new ContainerFetchEngineOptions
+        {
+            RetryInterval = TimeSpan.FromMilliseconds(1),
+            RetryLimit = 1,
+            ParallelDownloadLimit = 1,
+            GetFileAsyncTimeout = TimeSpan.FromMilliseconds(100),
+        };
+
+        private static readonly ContainerItem DummyConatinerItem1 = new ContainerItem
+        {
+            ItemType = ItemType.File,
+            FileLength = 52,
+            Path = "c:\\drop\\file1.txt"
+        };
+
+        private static readonly ContainerItem DummyConatinerItem2 = new ContainerItem
+        {
+            ItemType = ItemType.File,
+            FileLength = 52,
+            Path = "c:\\drop\\file2.txt"
+        };
+
+        [Fact]
+        [Trait("Level", "L0")]
+        [Trait("Category", "Worker")]
+        public async void ShouldDownloadAllTheFiles()
+        {
+            var stubContainerProvider = new StubContainerProvider(mockContainerItems, item => mockItemContent);
+            var fetchEngine = GetFetchEngine(stubContainerProvider);
+
+            Task fetchAsync = fetchEngine.FetchAsync();
+            await fetchAsync;
+
+            Assert.Equal(1, stubContainerProvider.GetItemsAsynCounter);
+            Assert.Equal(mockContainerItems, stubContainerProvider.GetFileTaskArguments);
+        }
+
+        [Fact]
+        [Trait("Level", "L0")]
+        [Trait("Category", "Worker")]
+        public async void ShouldHonorParallelDownloadLimit()
```

Comment:
Let's keep this specific to download limit of 1 (ShouldNotDoParallelDownloadIfSpecified)?